frontend/src/app/firewood/forests/buy-firewood-permit/buy-firewood-permit.component.ts

```diff
@@ -0,0 +1,263 @@
+    // this.applicationForm.get('orgStructureCode').setValue(forest.orgStructureCode);
+
+    if (this.permit) {
+      this.rePopulateForm();
+    }
+
+    this.applicationForm.get('numberOfCords').valueChanges.subscribe(value => {
+      this.quantityChange(value);
+    });
+  }
+
+  /**
+   * Redirect to tree guidelines if forest start date is after today
+   */
+  checkSeasonStartDate(forest) {
+    // if (forest && moment(forest.startDate).isAfter(moment())) {
+    //   this.router.navigate(['/christmas-trees/forests/', forest.forestAbbr]);
+    // }
+  }
+
+  /**
+  * handle the data of an existing application
+  */
+  handleData(isCancel) {
+
+    // this.firewoodInfoService.updateMarkdownText(this.markdownService, this.forest);
+
+    this.checkSeasonStartDate(this.forest);
+
+    // cancel any permits coming here that are still initiated and not yet completed
+    if (this.permit && isCancel && this.permit.status === 'Initiated') {
+      this.applicationService.updatePermit(this.permit.permitId, 'Cancelled', this.jwtToken).subscribe(updated => {
+        this.permit = updated;
+        this.showCancelAlert = true;
+      });
+    }
+
+    this.titleService.setTitle(
+      'Buy a Firewood permit | ' + this.forest.forestName + ' | U.S. Forest Service Open Forest'
+    );
+    this.createForm(this.forest, this.formBuilder);
+
+  }
+
+  /**
+   * Get data from route resolver
+   */
+  ngOnInit() {
+    this.winRef.getNativeWindow().location.hash = ''; // clear out the hash on reload
+
+    this.location.subscribe(locationChange => {
+      if (locationChange.type === 'hashchange') {
+        // back button press from #rules
+        this.showRules = false;
+      }
+    });
+    let isCancel = false;
+    this.route.queryParams.forEach((params: Params) => {
+      if (params.cancel)  {
+        isCancel = true;
+      }
+
+      if (params.t) {
+        this.jwtToken = params.t;
+      }
+    });
+
+    this.route.data.subscribe(data => {
+      if (data.forest) {
+        this.forest = data.forest;
+        this.permit = data.permit;
+        this.handleData(isCancel);
+      }
+    });
+  }
+
+  /**
+   * If form is submitted, show rules
+   * If permit already exists, redirect to permit view page
+   * If errors, scroll to error
+   */
+  showRulesForm() {
+    this.submitted = true;
+    this.showCancelAlert = false;
+    this.apiErrors = null;
+    this.applicationFieldsService.touchAllFields(this.applicationForm);
+    if (this.applicationForm.valid) {
+      this.showRules = true;
+      this.winRef.getNativeWindow().scroll(0, 200);
+      // const routeOptions = { fragment: 'rules' };
+      // if (this.permit) {
+      //   this.router.navigate(
+      //     [`/firewood/forests/${this.forest.forestAbbr}/applications`, this.permit.permitId],
+      //     routeOptions
+      //   );
+      // } else {
+      //   this.router.navigate([`/christmas-trees/forests/${this.forest.forestAbbr}/applications`], routeOptions);
+      // }
```

Comment:
Do we still need this or can we delete it?

Reply:
we will need this after we wire up the application piece